packages/stryker/test/unit/MutatorOrchestratorSpec.ts

```diff
@@ -5,20 +5,23 @@ import MutatorOrchestrator from '../../src/MutatorOrchestrator';
 import { Mutator, MutatorFactory, IdentifiedNode, Identified } from 'stryker-api/mutant';
 import * as sinon from 'sinon';
 import { Syntax } from 'esprima';
-import StrykerTempFolder from '../../src/utils/StrykerTempFolder';
+import { TempFolder } from '../../src/utils/TempFolder';
 import * as estree from 'estree';
 import StrictReporter from '../../src/reporters/StrictReporter';
 import { reporterStub } from '../helpers/producers';
+import { Mock, mock } from '../helpers/producers';
 
 describe('MutatorOrchestrator', () => {
   let sut: MutatorOrchestrator;
   let fileUtilsStub: sinon.SinonStub;
   let sandbox: sinon.SinonSandbox;
   let reporter: StrictReporter;
+  let tempFolderMock: Mock<TempFolder>;
 
   beforeEach(() => {
     sandbox = sinon.sandbox.create();
-    sandbox.stub(StrykerTempFolder, 'writeFile');
+    tempFolderMock = mock(TempFolder);
```

Comment:
Looks good! Awesome this works, even with a class with a private constructor.

Reply:
ç

Reply:
Ah true. Damn, i think this is a limitation of typescript. It will work if you you change the `mock`:

```typescript

export function mock<T>(constructorFn: { new(...args: any[]): T; }): Mock<T>;
export function mock<T>(constructorFn: any): Mock<T>;
export function mock<T>(constructorFn: any): Mock<T> {
  return sinon.createStubInstance(constructorFn) as Mock<T>;
}
```

And than use it like this: `mock<TempFolder>(TempFolder);`

Reply:
Hm. Now I am getting this: https://travis-ci.org/stryker-mutator/stryker/jobs/270886353#L1682